Catappult/appcoins-wallet-android
# [bugfix/APPC-3045] Promo code settings state

**What does this PR do?**

   fixes promo code state when opening the settings. If a user had an active promo code, the state was not updating so when it became expired, we were showing it as active on the settings

**Database changed?**

  No

**How should this be manually tested?**

 - Insert a active promo code (or ask for a new promo code for this test) and check that the correct state shows
 - Ask for MS to deactivate that promo code
 - recheck the state that appears in the settings

**What are the relevant tickets?**

  https://aptoide.atlassian.net/browse/APPC-3045

**Code Review Checklist**
